pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubit.hpp

```diff
@@ -632,5 +643,133 @@ class StateVectorLQubit : public StateVectorBase<PrecisionT, Derived> {
+        return distrib(rng);
+    }
+
+    /**
+     * @brief Calculate probabilities of measuring 0 or 1 on a specific wire
+     *
+     * @param wire Wire
+     * @return std::vector<PrecisionT> Both probabilities of getting 0 or 1
+     */
+    auto probs(const std::size_t wire) -> std::vector<PrecisionT> {
+        auto *arr = this->getData();
+        const size_t stride = pow(2, this->num_qubits_ - (1 + wire));
+        const size_t vec_size = pow(2, this->num_qubits_);
+        const auto section_size = vec_size / stride;
+        const auto half_section_size = section_size / 2;
+
+        PrecisionT prob_0{0.};
+        // zero half the entries
+        // the "half" entries depend on the stride
+        // *_*_*_*_ for stride 1
+        // **__**__ for stride 2
+        // ****____ for stride 4
+        for (size_t idx = 0; idx < half_section_size; idx++) {
+            const size_t offset = stride * 2 * idx;
+            for (size_t ids = 0; ids < stride; ids++) {
+                prob_0 += std::norm(arr[offset + ids]);
+            }
+        }
+
+        const std::vector<PrecisionT> probs{prob_0, PrecisionT(1.) - prob_0};
+        return probs;
+    }
+
+    /**
+     * @brief Collapse the state vector as after having measured one of the
+     * qubits
+     *
+     * @param wire Wire to collapse.
+     * @param branch Branch 0 or 1.
+     */
+    void collapse(const std::size_t wire, const bool branch) {
+        auto *arr = this->getData();
+        const size_t stride = pow(2, this->num_qubits_ - (1 + wire));
+        const size_t vec_size = pow(2, this->num_qubits_);
+        const auto section_size = vec_size / stride;
+        const auto half_section_size = section_size / 2;
+
+        // zero half the entries
+        // the "half" entries depend on the stride
+        // *_*_*_*_ for stride 1
+        // **__**__ for stride 2
+        // ****____ for stride 4
+        const size_t k = branch ? 0 : 1;
+        for (size_t idx = 0; idx < half_section_size; idx++) {
+            const size_t offset = stride * (k + 2 * idx);
+            for (size_t ids = 0; ids < stride; ids++) {
+                arr[offset + ids] = {0., 0.};
+            }
+        }
+
+        PrecisionT norm = std::sqrt(squaredNorm(arr, this->getLength()));
+
+        if (norm > std::numeric_limits<PrecisionT>::epsilon() * 1e2) {
+            std::complex<PrecisionT> inv_norm = 1. / norm;
+            for (size_t k = 0; k < this->getLength(); k++) {
+                arr[k] *= inv_norm;
+            }
+        }
+    }
+
+    /**
+     * @brief Measure one of the qubits and collapse the state accordingly
+     *
+     * @param wire Wire to measure.
+     * @param postselect Wire sample value to postselect (0 or 1) or -1 to
+     * ignore
+     * @param reset Flag to reset wire to 0
+     * @param branch Branch 0 or 1.
+     */
+    const auto measure(const std::size_t wire, const int postselect = -1,
```

Comment:
When we're done we could move this up in `applyMidMeasureMP` and delete `measure`. I think the Catalyst folks may then redefine the interface at their convenience.